app/src/main/java/ch/epfl/sdp/drone3d/ui/map/MissionInProgressActivity.kt

```diff
@@ -92,9 +95,52 @@ class MissionInProgressActivity : BaseMapActivity() {
         // TODO View stream
     }
 
+    private var speedObserver = Observer<Float> { speed ->
+        speedLiveText.apply {
+            text = getString(R.string.live_speed, speed.toString())
+        }
+    }
+
+    private var altitudeObserver = Observer<Float> { altitude ->
+        altitudeLiveText.apply {
+            text = getString(R.string.live_altitude, altitude.toString())
+        }
+    }
+
+    private var batteryObserver = Observer<Float> { batteryLevel ->
+        batteryLiveText.apply {
+            text = getString(R.string.live_battery, batteryLevel.toString())
+        }
+    }
+
+    private var distanceUserObserver = Observer<LatLng> { position ->
+        if (locationService.isLocationEnabled()) {
+            val distanceUser = position.distanceTo(locationService.getCurrentLocation()!!)
```

Comment:
Even though usually locationService.getCurrentLocation() is not null when locationService.isLocationEnabled() is true. It can append that getCurrentLocation is null if it is the first time you are trying to get the location from this service.

Reply:
Okay, but I already take that into account by using !! after getCurrentLocation() I think

Reply:
No, using !! specify that the program must crash if this is null.
What we should do is take care of the case where the location is null.